articat/catalog.py

```diff
@@ -12,6 +12,10 @@
 
 T = TypeVar("T", bound=Artifact)
 
+# Custom exception for missing artifacts
+class MissingArtifactException(ValueError):
```

Comment:
might be an overkill but I wouldn't mind a separate module for exceptions (even if it's just one for now).